jaeger-tracerresolver/src/main/java/com/uber/jaeger/tracerresolver/JaegerTracerResolver.java

```diff
@@ -0,0 +1,125 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.jaeger.tracerresolver;
+
+import com.uber.jaeger.Tracer;
+import com.uber.jaeger.metrics.Metrics;
+import com.uber.jaeger.metrics.NullStatsReporter;
+import com.uber.jaeger.reporters.RemoteReporter;
+import com.uber.jaeger.reporters.Reporter;
+import com.uber.jaeger.samplers.ConstSampler;
+import com.uber.jaeger.samplers.Sampler;
+import com.uber.jaeger.senders.Sender;
+import com.uber.jaeger.senders.UdpSender;
+
+import io.opentracing.contrib.tracerresolver.TracerResolver;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class JaegerTracerResolver extends TracerResolver {
+
+  /**
+   * Prefix for all properties used to configure the {@link TracerResolver}.
+   */
+  public static final String JAEGER_PREFIX = "JAEGER_";
+
+  /**
+   * The UDP maximum packet size used when communicating with the agent.
+   */
+  public static final String JAEGER_AGENT_UDP_MAX_PACKET_SIZE = JAEGER_PREFIX + "AGENT_UDP_MAX_PACKET_SIZE";
+
+  /**
+   * The UDP port used to locate the agent.
+   */
+  public static final String JAEGER_AGENT_UDP_PORT = JAEGER_PREFIX + "AGENT_UDP_PORT";
+
+  /**
+   * The UDP host used to locate the agent.
+   */
+  public static final String JAEGER_AGENT_UDP_HOST = JAEGER_PREFIX + "AGENT_UDP_HOST";
+
+  /**
+   * The maximum queue size for use when reporting spans remotely.
+   */
+  public static final String JAEGER_REPORTER_MAX_QUEUE_SIZE = JAEGER_PREFIX + "REPORTER_MAX_QUEUE_SIZE";
+
+  /**
+   * The flush interval when reporting spans remotely.
+   */
+  public static final String JAEGER_REPORTER_FLUSH_INTERVAL = JAEGER_PREFIX + "REPORTER_FLUSH_INTERVAL";
+
+  /**
+   * The service name.
+   */
+  public static final String JAEGER_SERVICE_NAME = JAEGER_PREFIX + "SERVICE_NAME";
+
+  static final int DEFAULT_REPORTER_FLUSH_INTERVAL = 500;
+  static final int DEFAULT_REPORTER_MAX_QUEUE_SIZE = 1000;
+
+  @Override
+  protected io.opentracing.Tracer resolve() {
+    String serviceName = getProperty(JAEGER_SERVICE_NAME);
+
+    return new Tracer.Builder(serviceName, getReporter(), getSampler()).build();
+  }
+
+  protected static Reporter getReporter() {
+    return new RemoteReporter(getSender(),
```

Comment:
Could we avoid writing this code and instead use the Configuration class to create the tracer? Here we just need to set the parameters on the respective Config classes, but leave the init logic to one place in the main config.